src/common/telemetryClient.ts

```diff
@@ -12,22 +12,31 @@ const aiKey: string = packageJSON.aiKey;
 
 export class TelemetryClient {
     public static sendEvent(eventName: string, properties?: { [key: string]: string; }): void {
-        if (properties) {
-            properties[Constants.isInternalPropertyName] = this._isInternal === true ? "true" : "false";
-        } else {
-            properties = {
-                [Constants.isInternalPropertyName] : this._isInternal === true ? "true" : "false",
-            };
-        }
+        this.stampInternalProperty(properties);
         this._client.sendTelemetryEvent(eventName, properties);
     }
 
+    public static sendErrorEvent(eventName: string, properties?: { [key: string]: string; }): void {
+        this.stampInternalProperty(properties);
+        this._client.sendTelemetryErrorEvent(eventName, properties, null, ["error", "errorMessage"]);

```

Comment:
Define a const list of error properties? And when using error properties, we could ref the consts.